pybamm/solvers/dae_solver.py

```diff
@@ -206,14 +207,113 @@ def eval_event(t, y):
+        )
+        y_casadi = casadi.vertcat(y_diff, y_alg)
+        pybamm.logger.info("Converting RHS to CasADi")
+        concatenated_rhs = model.concatenated_rhs.to_casadi(t_casadi, y_casadi)
+        pybamm.logger.info("Converting algebraic to CasADi")
+        concatenated_algebraic = model.concatenated_algebraic.to_casadi(
+            t_casadi, y_casadi
+        )
+        all_states = casadi.vertcat(concatenated_rhs, concatenated_algebraic)
+        pybamm.logger.info("Converting events to CasADi")
+        casadi_events = {
+            name: event.to_casadi(t_casadi, y_casadi)
+            for name, event in model.events.items()
+        }
+
+        # Create functions to evaluate rhs and algebraic
+        concatenated_rhs_fn = casadi.Function(
+            "rhs", [t_casadi, y_casadi], [concatenated_rhs]
+        )
+        concatenated_algebraic_fn = casadi.Function(
+            "algebraic", [t_casadi, y_casadi], [concatenated_algebraic]
+        )
+        all_states_fn = casadi.Function("all", [t_casadi, y_casadi], [all_states])
+
+        if model.use_jacobian:
+
+            pybamm.logger.info("Calculating jacobian")
+            casadi_jac = casadi.jacobian(all_states, y_casadi)
+            casadi_jac_fn = casadi.Function(
+                "jacobian", [t_casadi, y_casadi], [casadi_jac]
+            )
+            casadi_jac_alg = casadi.jacobian(concatenated_algebraic, y_casadi)
+            casadi_jac_alg_fn = casadi.Function(
+                "jacobian", [t_casadi, y_casadi], [casadi_jac_alg]
+            )
 
             def jacobian(t, y):
-                return jac.evaluate(t, y, known_evals={})[0]
+                return casadi_jac_fn(t, y)
+
+            def jacobian_alg(t, y):
+                return casadi_jac_alg_fn(t, y)
 
         else:
             jacobian = None
+            jacobian_alg = None
+
+        # Calculate consistent initial conditions for the algebraic equations
+        def rhs(t, y):
+            return concatenated_rhs_fn(t, y).full()[:, 0]
+
+        def algebraic(t, y):
+            return concatenated_algebraic_fn(t, y).full()[:, 0]
+
+        if len(model.algebraic) > 0:
+            y0 = self.calculate_consistent_initial_conditions(
+                rhs,
+                algebraic,
+                model.concatenated_initial_conditions[:, 0],
+                jacobian_alg,
+            )
+        else:
+            # can use DAE solver to solve ODE model
+            y0 = model.concatenated_initial_conditions[:, 0]
+
+        # Create functions to evaluate residuals
+
+        def residuals(t, y, ydot):
+            pybamm.logger.debug(
+                "Evaluating residuals for {} at t={}".format(model.name, t)
+            )
+            states_eval = all_states_fn(t, y).full()[:, 0]
+            return states_eval - model.mass_matrix.entries @ ydot
+
+        # Create event-dependent function to evaluate events
+        def event_fun(event):
```

Comment:
If events aren't working should we remove this bit of code?

Reply:
Still need this for setting up with casadi and passing to other solvers